### PR TITLE
fix(db): retrieve manifests before expensive step on prognosis

### PR DESCRIPTION
**Pull request by miguel-crespo-fdc**
Ref: SRX-YNEAHU

### CODE REVIEW

File: pkg/db/db.go [hunk 1]
```diff
@@ -602,6 +602,84 @@ func (h *DBHandler) DBSelectReleaseByVersion(ctx context.Context, tx *sql.Tx, ap
...
+func (h *DBHandler) DBSelectAllManifestsForAllReleases(ctx context.Context, tx *sql.Tx) (map[string]map[uint64]DBReleaseManifests, error) {
+	selectQuery := h.AdaptQuery(
+		`
+	SELECT DISTINCT
+		releases.appname,
+		releases.releaseVersion,
+		releases.manifests
+	FROM (
+		SELECT
+			MAX(releaseVersion) AS biggestRelease,
```
Comment (sven-urbanski-freiheit-com) [inline pkg/db/db.go:614]:
```suggestion
			MAX(releaseVersion) AS latestRelease,
```
(nitpick)

File: services/cd-service/pkg/repository/transformer.go [hunk 2]
```diff
@@ -3769,7 +3781,9 @@ func (c *envReleaseTrain) prognosis(
 		}
 
 		if state.DBHandler.ShouldUseOtherTables() {
-			release, err := state.DBHandler.DBSelectReleaseByVersion(ctx, transaction, appName, versionToDeploy, true)
+			release, exists := allLatestManifests[appName][versionToDeploy]
+			//release, err = state.DBHandler.DBSelectReleaseByVersion(ctx, transaction, appName, versionToDeploy, true)
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:3785]:
```suggestion
```

File: services/cd-service/pkg/repository/transformer.go [hunk 3]
```diff
@@ -3769,7 +3781,9 @@ func (c *envReleaseTrain) prognosis(
...
+
 			if err != nil {
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:3786]: this looks like it's never true

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/cd-service/pkg/repository/transformer.go (hunk 3 above)
Comment (miguel-crespo-fdc) [reply services/cd-service/pkg/repository/transformer.go:3786]: Looks like it, but we have no guarantee that it isn't. The old code checks for release == nil, so this is the equivalent to that

**Pull request merged**